kubernetes/apps/charts/metabase/templates/database.yaml

```diff
@@ -0,0 +1,142 @@
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: database
+  namespace: {{ .Release.Namespace }}
+  labels:
+    name: database
+spec:
+  replicas: {{ .Values.workloads.database.replicas }}
+  selector:
+    matchLabels:
+      name: database
+  serviceName: database
+  template:
+    metadata:
+      labels:
+        name: database
+      {{- with .Values.workloads.database.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+    {{- with .Values.images.database.imagePullSecrets }}
+      imagePullSecrets:
+      {{ toYaml . | nindent 6 }}
+    {{- end }}
+      containers:
+
+      - name: database
+        image: {{ .Values.images.database.src }}
+        {{- if .Values.configs.database }}
+        envFrom:
+        - configMapRef:
+            name: database-config
+        {{- end }}
+        ports:
+        - containerPort: {{ .Values.workloads.database.port }}
+          name: postgresql
+        {{- if .Values.volumes.database }}
+        volumeMounts:
+          {{- if .Values.volumes.database.data }}
+          - name: data
+            mountPath: /var/lib/postgresql/data
+            subPath: postgresql-data
+          {{- end }}
+        {{- end }}
+        readinessProbe:
+          tcpSocket:
+            port: {{ .Values.workloads.database.port }}
+          initialDelaySeconds: 5
+          periodSeconds: 10
+        livenessProbe:
+          tcpSocket:
+            port: {{ .Values.workloads.database.port }}
+          initialDelaySeconds: 15
+          periodSeconds: 20
```

Comment:
@lottspot I've found that it can be important for metabase to define a `startupProbe`, as metabase can take a long while to start up when cold sometimes and get stuck in a crash loop with k8s killing it for not being live fast enough. Defining a startupProbe lets you provide a longer threshold for initial readiness

See: https://github.com/urbanSpatial/LetsPlanOrg/blob/901dc868b44cc743dfc1a948820946034b80ded9/helm-chart/templates/metabase.yaml#L30

Reply:
Can you define "important"? What negative consequences are we concerned with if a metabase startup is in a crash loop until the database comes online? To me, this effect actually seems desirable-- if the app crashes when a DB connection isn't up, then it is 100% in a failed state, and it seems better for us to be aware of that than to engage in any attempt to mask it.

Reply:
furthermore, we're talking about a minuscule edge case here-- the only time metabase will be coming up when a DB is not up is in a fresh bootstrap scenario. This seems hardly worth holding up a merge over.

Reply:
@lottspot this isn't about metabase crashing until the database comes online, metabase is a fat cow of a java app that can take 30-60s plus to start up. On a slow or taxed enough node, metabase can get into a crash loop even with all its resources available because it doesn't start up fast enough to not get killed by k8s over and over before it can finish starting

Reply:
oh a case where this happens a LOT is when metabase has migrations to run (i.e. initial startup of a new metabase internal DB or during version bumps)

Reply:
feel free to merge and then add the startupProbe in a follow-up

Reply:
@themightychris gotcha-- I opened up #217 for continuation